downstream.go

```diff
@@ -1321,10 +1331,6 @@ func (dc *downstreamConn) sendTargetBacklog(net *network, target, msgID string)
 
 	dc.SendBatch("chathistory", []string{dc.marshalEntity(net, target)}, nil, func(batchRef irc.TagValue) {
 		for _, msg := range history {
-			if !dc.messageSupportsHistory(msg) {
-				continue
-			}
-
```

Comment:
Why is this `if` block removed.

IIUC this will break clients that do not support CHATHISTORY when the backlog contains anything other than PRIVMSG and NOTICE messages.

Reply:
Hm that seems correct...

Reply:
LoadLatestID only returns PRIVMSG & NOTICE messages by contract, so this check isnt needed

Reply:
I see, that makes sense.

Then `memoryMessageStore`'s `Append` should be modified to drop non-event messages (I think they are useless in this message store? otherwise they should be dropped at the `LoadLatestID` level)

https://github.com/emersion/soju/blob/832d8b89a2e6a13bf850ca7641a7203cbe64b413/msgstore_memory.go#L76